### PR TITLE
feat: allow `null` when subscribing notification

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/33632.

Allows for `systemPreferences.subscribe{Local|Workspace}Notification` methods to take `null` for their event parameters.

Per [documentation](https://developer.apple.com/documentation/foundation/nsdistributednotificationcenter/1414136-addobserver):

> When nil, the notification center doesn’t use a notification’s name to decide whether to deliver it to the observer.

cc @knev 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Enabled `systemPreferences.subscribe{Local|Workspace}Notification` to take a `null` value for the `event` parameter.

### CODE REVIEW

File: docs/api/system-preferences.md [hunk 1]
```diff
@@ -84,7 +84,7 @@ that contains the user information dictionary sent along with the notification.
 
 ### `systemPreferences.subscribeNotification(event, callback)` _macOS_
 
-* `event` string
+* `event` string | null
```
Comment (nornagon) [inline docs/api/system-preferences.md:87]: let's document what `null` does here

File: shell/browser/api/electron_api_system_preferences.h [hunk 2]
```diff
@@ -76,17 +76,17 @@ class SystemPreferences
   void PostNotification(const std::string& name,
                         base::DictionaryValue user_info,
                         gin::Arguments* args);
-  int SubscribeNotification(const std::string& name,
+  int SubscribeNotification(v8::Local<v8::Value> maybe_name,
```
Comment (nornagon) [inline shell/browser/api/electron_api_system_preferences.h:79]: `absl::optional` perhaps?

**Review by nornagon — COMMENTED** [review]
API LGTM, but have code / docs comments.

**Comment by arslan-raza-143** [comment]
> #### Description of Change
> 
> 
> 
> Closes https://github.com/electron/electron/issues/33632.
> 
> 
> 
> Allows for `systemPreferences.subscribe{Local|Workspace}Notification` methods to take `null` for their event parameters.
> 
> 
> 
> Per [documentation](https://developer.apple.com/documentation/foundation/nsdistributednotificationcenter/1414136-addobserver):
> 
> 
> 
> > When nil, the notification center doesn’t use a notification’s name to decide whether to deliver it to the observer.
> 
> 
> 
> #### Checklist
> 
> 
> 
> 
> 
> - [x] PR description included and stakeholders cc'd
> 
> - [x] `npm test` passes
> 
> - [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
> 
> - [x] relevant documentation is changed or added
> 
> - [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).
> 
> 
> 
> #### Release Notes
> 
> 
> 
> Notes: Enabled `systemPreferences.subscribe{Local|Workspace}Notification` to take a `null` value for the `event` parameter.

@codebytere

File: shell/browser/api/electron_api_system_preferences.h (hunk 2 above)
Comment (codebytere) [reply shell/browser/api/electron_api_system_preferences.h:79]: @nornagon `absl::optional` doesn't handle invalid values it handles missing values - so passing `null` with an optional string here would throw the same conversion error.

**Review by nornagon — APPROVED** [review]

**Review by miniak — APPROVED** [review]

**Comment by miniak** [comment]
API LGTM

**Comment by nornagon** [comment]
API LGTM

**Pull request merged**